### PR TITLE
Add post proposal: Can we teach Node Js without callbacks?

### PR DESCRIPTION
**Pull request by fmcarvalho**

### CODE REVIEW

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+tags: [web]
+author:
+  login: miguelgamboa
+  email: 
+  display_name: Miguel Gamboa
+  image: miguel.jpg
+  role: Software Developer
+---
+
+After another semester teaching Web Development the following question arises to
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: comma before "the following"

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+my mind: _can we teach Node Js without callbacks_?
+Considering that we cannot avoid Promises and async/await, so should we
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: consider removing the "so" before "should".

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+continually introduce callbacks in the first place? Notice that:
+1. Most part of Node API already supports Promises, e.g. file system module `fs`
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: Consider "Most Node APIs already..."

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+2. Starting lecturing with a simple use case of an HTTP client it is
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: Odd statement. No easy suggestion. Probably needs some restructuring.

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+   increasingly difficult to find a Node module that provides callback-based
+   API. One of those most used modules, i.e. `request` (with 23M downloads/week)
+   announced that is fully deprecated in February of 2020.
+
+Of course, we can always use the standard `http` module of Node, yet its API is
+far from being quite friendly.
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: Consider removing "quite".

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+2. Starting lecturing with a simple use case of an HTTP client it is
+   increasingly difficult to find a Node module that provides callback-based
+   API. One of those most used modules, i.e. `request` (with 23M downloads/week)
+   announced that is fully deprecated in February of 2020.
+
+Of course, we can always use the standard `http` module of Node, yet its API is
+far from being quite friendly. 
+
+Thus, this leads me to think that we could remove callbacks from the curricular
+plan of Web Development course. Can we?
```
Comment (pmhsfelix) [inline _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: Consider "the Web Development"

**Review by pmhsfelix — APPROVED** [review]
- Some comments to improve the reading.
- Meh.

File: _posts/2021-01-15-miguel-callbacks-yes-or-not.md (hunk 4 above)
Comment (fmcarvalho) [reply _posts/2021-01-15-miguel-callbacks-yes-or-not.md]: New proposal: _It is increasingly difficult to find an HTTP client Node module that provides callback-based API. For instance, one of those most used modules...._

**Pull request merged**